src/components/Notifications/ToastNotification/ToastNotificationList.tsx

```diff
@@ -201,6 +201,14 @@ const ToastNotificationList: FC<Props> = ({
     ? notifications.filter((notification) => filters.has(notification.type))
     : notifications;
 
+  useEffect(() => {
+    if (hasFilters && filteredNotifications.length === 0) {
+      // if there are no filtered notifications, reset the filters
+      setFilters(new Set());
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [filteredNotifications]);
```

Comment:
Not saying that this is better, but we could avoid silencing eslint (which I personally think we should not fall into the habit of doing) by removing the `hasFilters` check:
```suggestion
  useEffect(() => {
    if (filteredNotifications.length === 0) {
      // if there are no filtered notifications, reset the filters
      setFilters(new Set());
    }
  }, [filteredNotifications]);
```
I _think_ this should not cause an additional re-render when the Set is already empty, but I haven't tested it.
The alternative would be to have the default state set to `null` instead of the empty Set, so that we could do `setFilters(null)` here, and then it is much more likely that React will recognise that the state is already `null`, and so it won't re-render.

Anyway, this is just a tiny non-blocking consideration, so I'll approve and leave the final decision up to you.

Reply:
In the current implementation, `hasFilters` is missing from the dependency list of the use effect.

Reply:
> without it we are prone to reading an outdated value.

I'm honestly not sure how that could happen, and QA confirms that there are no such issues.

Anyway, I have updated my comment above to explain better why we should not include `hasFilters` in the dependency array. I hope it's clearer now.

Reply:
In the current form this is open for bugs where `hasFilters` is changed to `true` through a render, but the use effect still thinks it is `false`. You must define all reactive values inside of the dependency array of the use effect. See also https://react.dev/reference/react/useEffect#specifying-reactive-dependencies